libs/NfcCoreLib/lib/LibNfc/phLibNfc_Hci.c

```diff
@@ -1908,7 +1909,25 @@ NFCSTATUS phLibNfc_HciCreateApduPipeProc(void* pContext, NFCSTATUS status, void*
             pHciContext = (pphHciNfc_HciContext_t)pLibCtx->pHciContext;
             if (NFCSTATUS_SUCCESS == status)
             {
+                wStatus = status;
                 PH_LOG_LIBNFC_INFO_STR("Succeed to create APDU pipe");
+                phLibNfc_SE_Index_t seIndex = (pHciContext->hostApduPipeCreationNextSEIndex - 1);
+                phLibNfc_SE_List_t *seInfo = &pLibCtx->tSeInfo.tSeList[seIndex];
+                host = seInfo->hciHostId;
+                /*Check if the pipe was created for eSE */
+                if ((host >= phHciNfc_e_ProprietaryHostID_Min) &&
+                    (host <= phHciNfc_e_ProprietaryHostID_Max))
+                {
+                    /* Register events/responses coming to APDU Pipe */
+                    PH_LOG_LIBNFC_INFO_STR("Register for APDU pipe events.");
+                    phHciNfc_HciRegData_t tHciRegData;
+                    tHciRegData.eMsgType = phHciNfc_e_HciMsgTypeEvent;
+                    tHciRegData.bPipeId = pHciContext->aSEPipeList[PHHCI_ESE_APDU_PIPE_LIST_INDEX].bPipeId;
+                    (void)phHciNfc_RegisterCmdRspEvt(pHciContext,
+                        &tHciRegData,
+                        &phHciNfc_ProcessEventsOnApduPipe,
+                        pHciContext);
```

Comment:
`phHciNfc_RegisterCmdRspEvt` for `PHHCI_ESE_APDU_PIPE_LIST_INDEX` is already being called in `phLibNfc_HciGetSessionIdentityProc`.
Why do we need to call it here as well?

Reply:
For the very first Initialization after the NFCC FW download, Get session Id returns all 0xFFs.(As no pipes are present after FW download)
So during this time the events needs to be register at Pipe creation.
(Once pipes are created the Pipe ids will be set via Set Session Id, same can be retrieved by Get Session Id in next Initialization).
Once pipe is created its persistent till next FW download, so for all
subsequent Inits the registeration will be done through phLibNfc_HciGetSessionIdentityProc.
And during first Initialization(after FW download) via phLibNfc_HciCreateAPDUPipeProc.